src/withResolution.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import * as React from 'react'
+import { get } from 'lodash'
+import { displayAliases as defaultDisplayAliases } from './defaults.json'
+import type { DisplayValues } from './types'
+import { getMediaQueries } from './utils'
+import { ConfigContextPropTypes } from './configProvider'
+
+type Props = {
+  show?: DisplayValues,
+  base?: number,
+}
+
+type State = { shouldShow: boolean }
+
+export default function withResolution(Component: *) {
+  class WithResolution extends React.Component<Props, State> {
+    static contextTypes = ConfigContextPropTypes
+
+    constructor(props: Props) {
+      super(props)
+
+      this.state = {
+        shouldShow: true,
+      }
+      this.addMediaQueryListener(props.show)
+    }
+
+    componentWillReceiveProps({ show }: Props) {
+      this.addMediaQueryListener(show)
```

Comment:
this will remove and re-add the listener on every render?

Reply:
Good call, I can make it run only when `show` changes